baseapp/baseapp.go

```diff
@@ -654,12 +706,17 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 		anteEvents = events.ToABCIEvents()
 	}
 
-	// TODO remove nil check when implemented
-	if mode == runTxModeCheck && app.mempool != nil {
-		err = app.mempool.Insert(ctx, tx.(mempool.Tx))
+	if mode == runTxModeCheck {
+		err = app.mempool.Insert(ctx, tx)
 		if err != nil {
 			return gInfo, nil, anteEvents, priority, err
 		}
+	} else if mode == runTxModeDeliver {
```

Comment:
for my own understanding, why is it okay to not handle other types of `runTxMode`?

Reply:
We remove during `DeliverTx` (stated in ADR).